Pvt data update

denali49 opened the pull request:
- Documentation update

#### Description

backport to 2.2 branch.

JonathanLevi commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,23 +1,26 @@
+<!-- To uncheck box, add a space: [ ] -->
+<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+- [] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff).
+- [] I have added documentation to cover my changes or this change requires no new documentation.
```
Just to check: does the following suggestion make this clearer or worse (to a native English speaker) ?
I have **either** added documentation to cover my changes or this change requires no new documentation.

Otherwise, sorry in advance for my broken English ;-)

JonathanLevi commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,23 +1,26 @@
+- [] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff).
+- [] I have added documentation to cover my changes or this change requires no new documentation.
+- [] I have added unit tests to cover my changes or this change requires no new tests.
+- [] I have run [golint](https://github.com/golang/lint) and have fixed valid warnings in code I have added or modified. This tool generates false positives so you may choose to ignore some warnings. The goal is clean, consistent, and readable code.
```
Same Q as above here.

JonathanLevi commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,23 +1,26 @@
+<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+- [] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff).
+- [] I have added documentation to cover my changes or this change requires no new documentation.
+- [] I have added unit tests to cover my changes or this change requires no new tests.
```
Similarly:

I have **either** added unit tests to cover my changes or this change requires no new tests

JonathanLevi replied:
(To be clear: just asking)

christo4ferris commented:
I like the phrasing that @JonathanLevi suggested @srderson otherwise LGTM

srderson commented:
@JonathanLevi @christo4ferris Thanks for the feedback. I have updated the PR template with the suggested language.

christo4ferris commented:
LGTM thanks!

ghaskins commented:
LGTM

Commit:
[FAB-18028] Create new PropagateEnvironment key

Creates a new PropagateEnvironment to supersede the
existing key. If the new key is used, it supersedes
the existing key.

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>
(cherry picked from commit f12f80c3e77d638d176520740f386ab4714b2f22)

wlahti committed:
Disable channel participation API in release-2.2

FAB-18037 #done

Signed-off-by: Will Lahti <wtlahti@us.ibm.com>

Commit:
Add Check For CouchDB 3.1 (#1491)

Adds a warning to the log if the peer is using
a version of CouchDB prior to 3.1, as this is the
version we officially support.

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>
(cherry picked from commit 7fd87afa7ca160de835c6040a4af6fdebc40c06e)

Commit:
Upgrade updates

Change references to 2.x where applicable.

Fix language in channel update sequences.

Change-Id: Id6a02750f5cae393e44ac4e3317ec60ec6e896b0
Signed-off-by: joe-alewine <Joe.Alewine@ibm.com>
(cherry picked from commit bd00f8c2fb0e891cad3b90c2c9a1d05b06eac09b)

denali49 committed:
Deploy Chaincode Doc
Fix minor grammatical and typographical errors.

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>
(cherry picked from commit 09da681bfeb62daebd2200d180ae6c4ccb1c5966)

yacovm committed:
Only canonize ECDSA signatures in MSP:IsWellFormed (#1498)

Currently, the MSP IsWellFormed function expects any signature to be
a valid ECDSA signature, however the certificate can be signed by
a non-ECDSA algorithm which will then yield a false negative.

This change set ensures the check only applies if the signature is ECDSA.

Change-Id: I0b14e3e9b87e860a3ca29cc233dc4810de1768ab
Signed-off-by: yacovm <yacovm@il.ibm.com>

Commit:
Edits to style guide

Reorganized style guide into two sections.

Also added prohibition of whitelist, blacklist,
slave, master.

Change-Id: I3f8abb0285cd995669e6a61c5af949f1f4dcffa7
Signed-off-by: joe-alewine <Joe.Alewine@ibm.com>
(cherry picked from commit 1929e33b747202020512aba61999aa25ce820c5e)

denyeart committed:
Fix chaincode package example path

With the Fabric shift to go modules, chaincode packaging no longer
honors local GOPATH when resolving the chaincode path. User needs
to utilize a fully qualified path or a path relative to PWD.

This change updates the doc example that made an assumption about GOPATH usage.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit ebf1538468e7f79622990960dc5bb8950f6c1144)

manish-sethi committed:
Use directly leveldb batch (#1507)

This commit improves the leveldbhelper.UpdateBatch by
replacing the map that it maintains by the actual leveldb batch.

This reduces the memory consumed and a sorting of key-values is avoided
if the data added is already in sorted order (such as loading the data
from snapshot files)

Signed-off-by: manish <manish.sethi@gmail.com>

denyeart committed:
Add release notes for v2.2

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

Commit:
Remove Thirdparty Images From Bootstrap Script

Fabric 2.2 no longer uses the official hyperledger
couchdb. Fabric Samples has been shifted to use
the official images as well. As such, Samples
no longer relies on any of the images published
in the baseimage repo.

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>
(cherry picked from commit f52dc505c921ee755980c85b613e44f024760680)

sykesm committed:
Use tcp for statsd and poll for metric arrival

Have you heard the joke about UDP? Never mind. You probably won't get
it.

The integration tests seem to be flaking out while making assertions on
statsd metrics. There are two likely causes:

1. We wait for some sentinel and assume that when it arrives, all of the
   metrics we're looking for are ready. That's not necessarily true.
   This change replaces the single sentinel with assertions that all
   metrics are eventually available.
2. Even on good days and over the loopback adapter, UDP datagrams may
   not get to the intended target. To take that out of the equation,
   we'll move to TCP.

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>
(cherry picked from commit 74dd4dc4eb8ada43e36e73640b17bd6a0daab374)

denyeart committed:
Restore couchdb logger (#1532)

CouchDB client utility layer was recently refactored to be included in statecouchdb package.
This caused the traditional [couchdb] logger and [statecouchdb] logger
to log under a common [statecouchdb] logger.
Since administrators and support staff often want to see only the CouchDB interactions
in debug log, and since they are already accustomed to using the [couchdb] logger,
this change restores the separate [couchdb] logger for the CouchDB client code,
making it aligned with prior versions of Fabric.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

denyeart committed:
Add org name to consortium error message

Improve troubleshooting by including org name in the error message:
"Attempted to include a member which is not in the consortium"

New message:
"Attempted to include member BadOrgName which is not in the consortium"

This will help users that have an incorrect config transaction pinpoint the reason for failure.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit 5362a5701d0118629d6b974327555e4769e4232b)

denyeart committed:
Improve private data logging (#1545)

Changed following warning to debug message:
WARN The PvtRwset of PvtSimulationResultsWithConfig for txID [...] is nil. Skipping.
In v2.x peer attempts to retrieve private data from transient store one collection at a time.
If peer is eligible for multiple collections in a transaction,
and each transient store entry has one collection,
some of the retrieved records will have all collections filtered out and therefore the above message
is expected in v2.x and should not be a Warning (in v1.x this should not happen
since each transient store record was filtered against ALL eligible collections, therefore it
was a legitimate unexpected Warning message in v1.x).

Suppressed the following confusing message when there IS NO private data in a block:
INFO Successfully fetched all eligible collection private write sets for block [31]

When there IS private data in the block, replaced the message with a more useful message like this:
INFO Successfully fetched all 2 eligible collection private write sets for block [35] (0 from local cache, 1 from transient store, 1 from other peers)

And if peer can't fetch all the private data in a block, you'll now get this message:
WARN Could not fetch all 2 eligible collection private write sets for block [36] (0 from local cache, 1 from transient store, 0 from other peers).
     Will commit block with missing private write sets.

Finally, removed some of the redundant or not useful debug messages.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit 9d08ec7ab432dc127b44350f8ef819bb5ed6376e)

caod123 committed:
[FAB-18054] Remove default value for top level orderer addresses if (#1550)

unset in config

* Top level orderer addresses are deprecated in 2.2 so this commit
removes the default value for consumers that pass a config with no
orderer addresses value

Signed-off-by: Danny Cao <dcao@us.ibm.com>
(cherry picked from commit ff39bf9b4df7aa5fa24205c8b77ec5ad4038decd)

denyeart committed:
Improve peer CLI error message when no orderer address passed (#1555)

For peer CLI commands that require an orderer address, the peer
does best effort attempt to retrieve orderer address from channel
config if the orderer address is not passed (it currently only
checks the deprecated Orderer.Addresses, not the org level
OrdererEndpoints). The best effort attempt may be removed in
a future release, so for now improve the error message
to indicate that orderer address should be passed using
-o flag.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit 6ef8f3dfd2aed90c94b9fe4f8077834fa7b853ba)

denyeart committed:
Fabric v2.2.0 release commit

Update docs, change log, and scripts for v2.2.0 release.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

wlahti committed:
Remove ccenv dependency in platforms unit test

Signed-off-by: Will Lahti <wtlahti@us.ibm.com>
(cherry picked from commit a817f89d666dce69dc06f409694e6c6ee9e51d1b)

ODOWDAIBM committed:
Changes for i18n

Signed-off-by: Anthony O'Dowd <a_o-dowd@uk.ibm.com>
(cherry picked from commit a3be61aa2b22d0539dc919c7fab8399cae5f8bdd)

ibrandis committed:
Simple Typo fix

Signed-off-by: Ibrandis <ibrandis.dov@gmail.com>
(cherry picked from commit 98277ad667a428f019350c67a46da79cbe422945)

yacovm committed:
Print channel name in learnAnchorPeers

This change set adds printing of the channel name the anchor peers
are part of, in the function learnAnchorPeers.

This change set adds printing of the channel name the anchor peers
are part of, in the function learnAnchorPeers.

Change-Id: I00b8d6a0c40254c943b96bee468a9e19a8270ef2
Signed-off-by: yacovm <yacovm@il.ibm.com>

ODOWDAIBM committed:
Advice for translators, PRs and tools

Signed-off-by: Anthony O'Dowd <a_o-dowd@uk.ibm.com>
(cherry picked from commit 709d3d7dddca9b616b9cf15eff4311db9b43865d)

manindra23 committed:
Correct the explanation for signcerts in Membership section

Signed-off-by: Manindra Singh <manindra23@yahoo.co.in>
(cherry picked from commit 5a372b9c0ca17707c50f2bb9436f7494021924ea)

denyeart committed:
Add LTS release status to What's New doc.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit 5ce84827c9e09c85f8cc56b42ca361d8df2b3bf7)

Commit:
Raft: Check suspect info once per suspect interval (#1600)

Today's existing suspect logic has a periodic checker, which checks
every 10s if the Raft cluster still has quorum.  If the cluster has lost
quorum, it marks the time this event begins, then, every 10s checks to
see if 'enough' time has elapsed since the quorum was lost to suspect
that the OSN has been evicted.

If the OSN has not been evicted, or cannot determine its eviction
status, then every 10s the OSN attempts to re-check its suspicion
status, which can lead to large volumes of network traffic, especially
in significiantly multichannel environments.

This commit modifies the logic to track the number of times that the
suspect checking logic has actually executed, to ensure that we check no
more than once every suspect interval (by default every 10m, instead of
every 10s).

Signed-off-by: Jason Yellick <jyellick@us.ibm.com>
(cherry picked from commit c90015c9bfe9c22896e9187ca4f0d5975acb2305)

Commit:
Add link to Use a CA to deployment guide

The deployment guide overview was written before the
Use a CA doc was merged. This change adds a link to that
doc and cleans up a bit of the language in the appropriate
section.

Change-Id: I15605fe498ac8a3057724f0f63af71cbc9667ddb
Signed-off-by: joe-alewine <Joe.Alewine@ibm.com>
(cherry picked from commit ab3d8c1cf763bbfe52a841091df790e93a0fa32c)

cendhu committed:
fix missing err check in the block commit path (#1543)

Signed-off-by: senthil <cendhu@gmail.com>

ODOWDAIBM committed:
Fix international language make html instructions

Signed-off-by: Anthony O'Dowd <a_o-dowd@uk.ibm.com>
(cherry picked from commit 4c427b4ff57f97393e7a5cb922b7de5e36916fca)

sebastiantf committed:
Fix: Missing addToWallet.js in ls command output

Signed-off-by: Sebastian T F <sebastian.tharakan97@gmail.com>
(cherry picked from commit 6b4ca051ee46b0249c2da7f4d0303853282a4cf3)

Commit:
Fix link in Develop apps doc

Link to Transaction flow doc was broken

Change-Id: I139dcb028a3bee419ee0537c5d0541d9c5e11b05
Signed-off-by: joe-alewine <Joe.Alewine@ibm.com>
(cherry picked from commit af0d871b3b3cf9915ed023c0d013b1ef0721ef12)

zhaojianpeerfintech committed:
fix: modify certificate alternate DNS duplication

Signed-off-by: zhj0811 <zhaojian@peerfintech.cn>
Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

nhoriguchi committed:
fix glossary.rst by reverting commit 21fa6480b161

Commit 21fa6480b161 converted the sentence "the latter would tell the
former about ``peer0.orgA``".  But this paragraph explains with example
that the info about peer0.orgA is transferred to peer1.orgB (referred to
as "the former") via peer0.orgC (referred to as "the latter").  So the
original text is totally correct. Let's revert the commit.

Signed-off-by: Naoya Horiguchi <naoya.horiguchi@nec.com>
(cherry picked from commit 1e2ecf206e2b2f8007f8511fdc4b12a15f996fb4)

billykern committed:
typo fix

Signed-off-by: Billy Kern <william.j.kern@gmail.com>
(cherry picked from commit 9c705811e9d2dbda6a7876d0065d65a369a62ca5)

Commit:
Log Error When PKCS11 Key Not Found (#1679)

We return a series of errors if we are unable to
find a key in an HSM. We explicitly check if the error
was nil, but never log the error for debugging before
falling back to software.

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>

yacovm committed:
[FAB-18208] Do not sign gossip message if membership is empty

This change set makes gossip discovery and channel modules only sign (and gossip)
alive messages and state info messages if the membership is non empty.

Change-Id: Id4de507cdccaa5860ba4b67cf059defc98545bd9
Signed-off-by: yacovm <yacovm@il.ibm.com>

stephyee committed:
Updated Using CouchDB to use asset transfer ledger queries smart contract

Signed-off-by: Tiffany Harris <tiffany.harris@ibm.com>
(cherry picked from commit 7f8a63320ea5cf93d84ef5d010d4b50fff3f5f31)

denyeart committed:
Updates to CouchDB doc tutorial

Minor updates based on the refactored tutorial that
uses the new asset transfer ledger queries chaincode.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit 216fa9f4c2e56b02c080564ffbd503fc3137923c)

cendhu committed:
minor cleanup of pvtdatastore

Signed-off-by: senthil <cendhu@gmail.com>

yacovm committed:
Use correct gossip SecretEnvelope reference

In case a membership request is received, it contains an inner
alive message envelope (which has a secret envelope).

The function handleMsgFromComm extracts an internal endpoint from
that secret envelope, however it was referencing the wrong envelope.

Instead of referencing the inner envelope of the alive message, it referenced
the outer envelope of the membership request, however, only alive messages
carry internal endpoints inside their secret envelopes.

As a result, all internal endpoints were always empty.

While this has no semantic effect, because the consumer of the internal endpoint doesn't
make use of the field, the field might still be logged, and we should use the correct
envelope which contains the internal endpoint and not the incorrect envelope.

Change-Id: I2452269e627e3444485425b0df5aced527d5149c
Signed-off-by: yacovm <yacovm@il.ibm.com>

cendhu committed:
construct unreconciled missing pvtdata (#1699)

Signed-off-by: senthil <cendhu@gmail.com>

Commit:
Add Raft metrics to Raft config doc

Adding two metrics to Raft configuration doc to encourage
users to monitor their active nodes as compared to the
total number of nodes in the cluster

Change-Id: Ica35ee1a0a2feea6511cfd56fcee1bdf429b1334
Signed-off-by: joe-alewine <Joe.Alewine@ibm.com>
(cherry picked from commit 4c2ad49cb3d02655059f25e1d0bfa3e1901269fe)

stephyee committed:
Update Channel Update (Adding an Org) tutorial

Switches the tutorial to use the asset-transfer-basic sample instead of
the fabcar sample

Signed-off-by: Tiffany Harris <tiffany.harris@ibm.com>
(cherry picked from commit 4fae6eda9bd72b68cab6413a024caa5fb0a638bf)

sykesm committed:
Merge pkcs11/impl.go and pkcs11/pkcs11.go

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>
Signed-off-by: Gari Singh <gari.r.singh@gmail.com>

sykesm committed:
Replace loadLib with initialize method

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>
Signed-off-by: Gari Singh <gari.r.singh@gmail.com>

sykesm committed:
Make findKeyPairFromSKI a method on impl

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>
Signed-off-by: Gari Singh <gari.r.singh@gmail.com>

sykesm committed:
Make ecPoint a method on impl

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

cendhu committed:
refactor pvtdatastore

Signed-off-by: senthil <cendhu@gmail.com>

cendhu committed:
mv oldpvtdata commit to separate file

Signed-off-by: senthil <cendhu@gmail.com>

sykesm committed:
Add object handle cache to PKCS#11 bccsp provider

With this change, object handles obtained through findKeyPairFromSKI are
cached in the PKCS#11 implementation of BCCSP.

In support of this change, in addition to pooling idle sessions, the
provider tracks active sessions. If some condition occurs that results
in all sessions being closed, cached object handles are no longer valid
so the handle cache is purged.

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
Cache bccsp keys generated from getECKey

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>
Signed-off-by: Gari Singh <gari.r.singh@gmail.com>

sykesm committed:
Drain session pool before creating new sessions

If a bad session was retrieved from the session pool, a new session
would be immediately created instead of attempting to retrieve another
session from the pool.

This change attempts to uses pooled sessions before creating new ones.

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>
Signed-off-by: Gari Singh <gari.r.singh@gmail.com>

pamandrejko committed:
[FAB-18109] Update peer chaincode invoke

Need to describe how to target a specific smart contract inside
a chaincode package.
- Added a new example to peer chaincode invoke
- Added instructions to writing your first application tutorial.

Content is already included in the Developing Apps documentation.

Signed-off-by: pama-ibm <pama@ibm.com>
(cherry picked from commit 724f46425e37982cc992d4003459c498767930ab)

denyeart committed:
Update Add an Org to a Channel tutorial

A few minor updates:

- Indicate that NodeOU admin is now utilized rather than admincerts.
- Indicate that leader election is not the default starting in v2.2.
- Fix the endorsement to use Org2 and the new Org3 peer (2 out of 3 required).

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit 2837fc5cad7c36439c42e2ccfbdcf88a6dd61e97)

mergify[bot] committed:
FAB-18163 TLS Timeshift w/o Separate Cluster Port (#1724) (#1725)

The existing orderer code allows for a TLS timeshift to be specified
when starting the orderer with a separate cluster port.  This is useful
for situations where the consenter TLS certificates have accidentally
expired, preventing quorum from forming, and preventing channel config
updates to replace these certificates.

This change extends the concept to the general orderer port, which can
be optionally re-used, rather than binding to a new port.  If the
cluster port is not re-used, then the TLS timeshift parameter specified
in the general TLS section takes precdence over the parameter specified
in the cluster section.  Likewise, if there is a separate cluster port,
the parameter in the general TLS section has no impact on the cluster
listener.

Signed-off-by: Jason Yellick <jyellick@us.ibm.com>
(cherry picked from commit 5d071868a9f91236a103b285f95120e1e8523e94)

Co-authored-by: Jason Yellick <jyellick@us.ibm.com>

mergify[bot] committed:
Update doc to reflect change in FAB-18163 (#1729)

Signed-off-by: Jason Yellick <jyellick@us.ibm.com>
(cherry picked from commit 9881a25f43bdf39436bd0eff1449d65a495dac29)

Co-authored-by: Jason Yellick <jyellick@us.ibm.com>

yacovm committed:
Fix and improve discovery TLS authentication comments in document

Change-Id: I9a39ab88244438ccfb8ce97c235c45ec1c421419
Signed-off-by: yacovm <yacovm@il.ibm.com>
(cherry picked from commit ea7ab2d247c4211963db0ab72c9d2674523cccbe)

DereckLuo committed:
[FAB-18169] Add DevMode support in ChaincodeEndorsementInfoSource

* Added DevMode support in ChaincodeEndorsementInfoSource, when
  chaincode is launched by developer and missing Installinfo from
  chaincodeInfo cache, rebuild Installinfo and return accordinly.

Signed-off-by: Chongxin Luo <Chongxin.Luo@ibm.com>

DereckLuo committed:
[FAB-18120] Adding DevMode integration test for new lifecycle.

* Added integration test for DevMode using new lifecycle.

Signed-off-by: Chongxin Luo <Chongxin.Luo@ibm.com>

denali49 committed:
Refactor tutorial to 'Writing Your First Chaincode'

Signed-off-by: Chris Gabriel <chris@hyperchainlabs.com>

Address comments by maintainers

Signed-off-by: Chris Gabriel <chris@hyperchainlabs.com>
(cherry picked from commit 29e657102b1f06bdb1f692ab355e3a1899f563d5)

Commit:
Update RTD to Target Correct GH Release Branch

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>

Commit:
Remove Use of Manifest Tool

Since we no longer publish multiple architectures to
Dockerhub, simplify the release process by pushing images
directly with Docker.

As Docker currently has an experimental utility as part
of the Docker CLI: https://docs.docker.com/docker-for-mac/multi-arch/
future multiarch implementation could make use of the `buildx`
tooling directly from Docker assuming it is out stable.

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>
(cherry picked from commit 75d132b5683e7456857bb6758781b7eeec1b15ca)

mergify[bot] committed:
Update Write Your First Application Tutorial (#1754)

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

Remove reference to FabCar

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

Fix typo, address comments

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

Update to address requested reviewer changes

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

Fix case and grammatical errors

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>
(cherry picked from commit 647f0831c43cfe92619f03a3c1682de6d6ad7693)

Co-authored-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

mergify[bot] committed:
Update "Using the Fabric Test network" tutorial (#1755)

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

Update to reflect changes to deployCC script.

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

Include updated script example usage.

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>
(cherry picked from commit 73bdfcbda65ead3a479532f2a182e687b777626d)

Co-authored-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

mergify[bot] committed:
Update "Deploying a smart contract" tutorial. (#1756)

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>
(cherry picked from commit 4cb564f17d59c1f572f8caa98e7c8e20d8d6cf57)

Co-authored-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

stephyee committed:
Update docs to replace fabcar references with basic asset transfer
sample

Signed-off-by: Tiffany Harris <tiffany.harris@ibm.com>
(cherry picked from commit 87adee2b2e9232e7cf99fe357e6607195b1dd267)

mergify[bot] committed:
Write Your First App tutorial updates (#1757)

Update Write Your First App tutorial based on latest
javascript application.
Also includes misc tutorial updates and clarifications.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit e1729dfbc36da1f4cfdb5366084b0fc37e4b5b9b)

Co-authored-by: David Enyeart <enyeart@us.ibm.com>

mergify[bot] committed:
Fix code snippet display (#1759)

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

Update 'A closer look' section to reflect changes to application-javascript

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

Fix typo

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>
(cherry picked from commit ce5a4c78e34163343c8787c3de3b2927977660ff)

Co-authored-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

denali49 committed:
Add links for Go and Java sample applications

Signed-off-by: Chris Gabriel <chris@hyperchainlabs.com>

Update links to specify master branch path

Signed-off-by: Chris Gabriel <chris@hyperchainlabs.com>
(cherry picked from commit 8be3ebd70a4967d6464958aa769e192d271bf334)

yacovm committed:
[FAB-18171] Disregard certificate validity period in intra-orderer communication

This change set makes the orderer cluster authentication infrastructure
disregard validity periods when comparing certificates, and only regard public keys.

With this change, one can replace the TLS certificate of Raft,
by a certificate that has the same public key without issuing channel config updates.

This is to ensure that if a certificate of a consenter expired,
one can quickly renew it and start the orderer without performing
a config update per channel.

Change-Id: Id1b29e220d37aa33617b2143b702075bf5b01f6f
Signed-off-by: yacovm <yacovm@il.ibm.com>

pamandrejko committed:
Add instructions for how to use @Mergifyio backport command

Signed-off-by: pama-ibm <pama@ibm.com>
(cherry picked from commit 85ca436646cb8aaa7d60291e18aa96e2c066823a)

satota2 committed:
[FAB-18183] Bump sphinx in requirements.txt to v1.8.5

This patch bumps sphinx to v1.8.5 to use the same version as Pipfile.

Signed-off-by: Tatsuya Sato <Tatsuya.Sato@hal.hitachi.com>
(cherry picked from commit 57f2834d1425f2b918f62ed77a0adc40d2d4d91e)

denyeart committed:
[FAB-18188] Log orderer and peer cert expiration date upon startup (#1804)

FAB-17000 added a warning if expiration is within a week.

This change additionally info logs the future enrollment, tls server, and tls client
cert expiration dates upon startup, regardless of whether they will expire within a week or not.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit bd7180cd1be1e20b6a48b356e093afaddab082cb)

Commit:
Revert "[FAB-18183] Bump sphinx in requirements.txt to v1.8.5"

This commit created a formatting issue in the doc.

This reverts commit 445b997f66bcbda77a5e2ebe18919cb1d09c7a26.

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>

cendhu committed:
deprioritize unreconcilable missingPvtData (#1721)

* deprioritize unreconcilable missingPvtData

Signed-off-by: senthil <cendhu@gmail.com>

* address review comment

Signed-off-by: senthil <cendhu@gmail.com>

* add review comments

Signed-off-by: senthil <cendhu@gmail.com>

* address review comments

Signed-off-by: senthil <cendhu@gmail.com>

wenjianqiao committed:
[FAB-18191] Remove contents of leveldb dir instead of the dir itself when dropping dbs (#1828)

When dropping a leveldb, remove all contents of the directory instead of
the directory itself. This is common code for upgrade-dbs, rebuild, reset, and rollback.

Signed-off-by: Wenjian Qiao <wenjianq@gmail.com>

Commit:
Update RTD Placeholder

Dynamically pull the RTD version from the environment
variable delivered by RTD

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>

cendhu committed:
pass unreconciled missing data info to ledger from reconciler (#1797)

Signed-off-by: senthil <cendhu@gmail.com>

denyeart committed:
[FAB-18194] Fix service discovery for legacy installed chaincodes

Service discovery endorsers query fails with errors:
"failed constructing descriptor for chaincodes"
"required chaincodes are not installed on sufficient peers"

The chaincodes installed with legacy chaincode lifecycle were not getting recognized
after peer restart since the path to installed chaincodes was set after
the legacy chaincode cache initialized.
Setting the installed chaincode path earlier in peer start sequence resolves the issue.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
(cherry picked from commit 0c10c0ad5272dfe59efac918bb1b301649832950)

cendhu committed:
reset leveldb batch after the commit

While committing data to leveldb in iteration, it is
necessary to reset the batch. Otherwise, we would be
appending entries and rewriting written entries in the
previous iteration.

Signed-off-by: Senthil Nathan N <cendhu@gmail.com>

yacovm committed:
Fix data race in gossip/discovery test

This change set fixes two data races:

1) A logger reference was overriden while the logger might be in use.
   I changed the implementation so that the logger will be injected.

2) A shared number was incremented by logger hooks that are instantiated
   multiple times for different peers.
   I made it so that each logger hook receives its own counter, which is
   no longer a counter but a map of log entries that are searched.

Change-Id: Ic69f604f30a16e1a1fb9050ba9145dfa38e05146
Signed-off-by: yacovm <yacovm@il.ibm.com>

denali49 committed:
Minor doc fix to clear Sphinx local build error

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>
(cherry picked from commit 20bd1035043251443744ef3c4afaf29ccd03856c)

Commit:
Convert Azure Pipeline To Stages (#1874)

This is the new model AZP will use going forward.
But it also allows you to retrigger failed stages
before other stages have completed. Thus, if unit
test fails early, or integration test, you can rerun
the stage without having to wait for other stages to
complete.

Signed-off-by: Brett Logan <brett.t.logan@ibm.com>

denali49 committed:
Update 'Using Private Data in Fabric' tutorial

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

fix invoke command on first ReadAsset

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

minor grammatical fixes

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

resolve comments

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

Add Asset Transfer section header and delete pipfile

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

address comments and add use case section

Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

wlahti committed:
Peer CLI communicate with orderers with expired TLS certs (#1863)

Implement a TLS handshake timeshift for the "peer channel fetch"
and "peer channel update" comands to allow fetching config blocks
and updating the config for orderers with expired TLS certificates.

FAB-18205 #done

Signed-off-by: Will Lahti <wtlahti@us.ibm.com>

wlahti committed:
Bump fabric-config dep to 0.0.7

Signed-off-by: Will Lahti <wtlahti@us.ibm.com>

Commit:
Adding notes for the usage of script during samples install

Signed-off-by: Rijul Aggarwal <rijul.aggarwal@ibm.com>
(cherry picked from commit 4ce4e10b36d6015d3921b84a80b9740464125b87)

wlahti committed:
Regenerate peer CLI docs

Signed-off-by: Will Lahti <wtlahti@us.ibm.com>

denyeart committed:
pass unreconciled missing pvtdata to pvtdata store (backport to release-2.2) (#1886)

* add config for deprioritized missing data access interval

Signed-off-by: Senthil Nathan N <cendhu@gmail.com>

* minor refactoring of missing pvtdata retrieval test

Signed-off-by: Senthil Nathan N <cendhu@gmail.com>
Signed-off-by: David Enyeart <enyeart@us.ibm.com>

* pass unreconciled data to the pvtstore commit

Signed-off-by: Senthil Nathan N <cendhu@gmail.com>

Co-authored-by: Senthil Nathan N <cendhu@gmail.com>

jebediahelliott committed:
Update write_first_app.rst

Fix typo for installation of `build-essential` package

Signed-off-by: Jeb Elliott <jelliott@andculture.com>

denyeart committed:
address review comments (#1890) (#1893)

Signed-off-by: Senthil Nathan N <cendhu@gmail.com>
Signed-off-by: David Enyeart <enyeart@us.ibm.com>

Co-authored-by: Senthil Nathan N <cendhu@gmail.com>

wlahti committed:
Clarify tlsHandshakeTimeShift CLI help text (#1894)

It only applies to handshakes with the orderer endpoint.

FAB-18205

Signed-off-by: Will Lahti <wtlahti@us.ibm.com>

Salmandabbakuti committed:
fix function name typo in store private data command

Signed-off-by: Salmandabbakuti <Salmandabbakuti@github.com>

mergify[bot] committed:
Bugfix in collection config history mgr (#1904)

Use proper stop key to prevent iterator returning results from next namespace

Signed-off-by: manish <manish.sethi@gmail.com>
(cherry picked from commit d67feb1e478f477d9d8572ee9b47d63111ef151a)

Co-authored-by: manish <manish.sethi@gmail.com>

nikhil550 reviewed (commented):
200 Files in this PR. Maybe manually copy the file might be better? 

I was going to make some edits to the tutorial, so not sure it if necessary to cherry pick this right away.

denali49 commented:
> 200 Files in this PR. Maybe manually copy the file might be better?
> 
> I was going to make some edits to the tutorial, so not sure it if necessary to cherry pick this right away.

Was trying to backport merged PR to 2.2 and got a failed merge message.  Executed the manual cherry-pick process documented here https://hyperledger-fabric.readthedocs.io/en/release-2.2/github/github.html#cherry-picking-your-pr-to-other-branches
and this is what happened.  Not intentional.

denali49 commented:
I’m going to close this PR Nik. 

Sent from my iPhone

> On Sep 21, 2020, at 12:27 PM, Chris Gabriel <notifications@github.com> wrote:
> 
> ﻿
> 200 Files in this PR. Maybe manually copy the file might be better?
> 
> I was going to make some edits to the tutorial, so not sure it if necessary to cherry pick this right away.
> 
> Was trying to backport merged PR to 2.2 and got a failed merge message. Executed the manual cherry-pick process documented here https://hyperledger-fabric.readthedocs.io/en/release-2.2/github/github.html#cherry-picking-your-pr-to-other-branches
> and this is what happened. Not intentional.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
